### PR TITLE
Feature: OVMwasm darwin support (x86_64, arm64)

### PR DESCRIPTION
**Pull request by judah-caruso**
This PR adds mac support for libovmwasm (#48) with/without libdyncall. Linux should be tested to ensure these changes didn't affect its build process.

### CODE REVIEW

File: interpreter/src/vm/vm.c [hunk 1]
```diff
@@ -204,7 +219,8 @@ bool ovm_engine_memory_ensure_capacity(ovm_engine_t *engine, i64 minimum_size) {
     if (engine->memory == NULL) {
         new_addr = mmap(NULL, minimum_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
     } else {
-        new_addr = mremap(engine->memory, engine->memory_size, minimum_size, MREMAP_MAYMOVE);
+        munmap(engine->memory, engine->memory_size);
+        new_addr = mmap(NULL, minimum_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
```
Comment (judah-caruso) [inline interpreter/src/vm/vm.c:223]: Does this maintain the same behavior as `mremap`?

Comment (brendanfh) [reply interpreter/src/vm/vm.c:223]: I messed up and deleted the wrong thing... So I will retype my comment from memory.

I don't think that is the same thing. That will lose all current memory, and replace it with with a fresh set of zeroed-pages.
I think the closest solution would be:
```c
new_addr = mmap(engine->memory, minimum_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
if (new_addr == MAP_FAILED) {
    // Panic
    return false;
} else if (new_addr != engine->memory) {
    // Mapped region has moved, need to copy everything
    memcpy(new_addr, engine->memory, engine->memory_size);
    munmap(engine->memory);
} else {
    // Mapped more memory with the same start pointer, do nothing
}
```

Comment (brendanfh) [reply interpreter/src/vm/vm.c:223]: However, I think still using `mremap` on Linux would be better because it is guaranteed to work and be fast.

Comment (judah-caruso) [reply interpreter/src/vm/vm.c:223]: 🤦 can't believe I forgot the memcpy... Will update this

Comment (judah-caruso) [reply interpreter/src/vm/vm.c]: Fixed and tested. Created  program that would segfault with the original changes, no longer does!

Comment (brendanfh) [reply interpreter/src/vm/vm.c]: Awesome!

**Comment by brendanfh** [comment]
The Linux builds look like they are still compiling, so it looks good to me. I will have to add the build configurations to the GitHub actions to build the x86_64 and arm64 versions automatically, but I will do that in a separate branch and pull-request.

Thanks for the great work! :tada:

**Pull request merged**